node/pkg/guardiansigner/benchmarksigner.go

```diff
@@ -0,0 +1,59 @@
+package guardiansigner
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"time"
+)
+
+type BenchmarkSigner struct {
+	innerSigner GuardianSigner
+}
+
+func NewBenchmarkSigner(unsafeDevMode bool, signerKeyPath string) (*BenchmarkSigner, error) {
+	innerSigner, err := NewGuardianSignerFromUri(signerKeyPath, unsafeDevMode)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to create benchmark signer: %w", err)
+	}
+
+	return &BenchmarkSigner{
+		innerSigner: innerSigner,
+	}, nil
+}
+
+func (b *BenchmarkSigner) Sign(hash []byte) ([]byte, error) {
+
+	start := time.Now()
+
+	sig, err := b.innerSigner.Sign(hash)
+
+	duration := time.Since(start)
+	fmt.Printf("Signing execution time: %v\n", duration)
+
+	return sig, err
+}
+
+func (b *BenchmarkSigner) PublicKey() ecdsa.PublicKey {
+
+	start := time.Now()
+
+	pubKey := b.innerSigner.PublicKey()
+
+	duration := time.Since(start)
+	fmt.Printf("Public key retrieval time: %v\n", duration)
```

Comment:
This could be exposed through [a Prometheus histogram](https://prometheus.io/docs/practices/histograms/).

Reply:
Here's an example of the definition of an histogram https://github.com/wormhole-foundation/wormhole/blob/02f468f776f6c8b8d20c1cb5d192818c4fc20c75/node/pkg/processor/processor.go#L169 and how to observe latency https://github.com/wormhole-foundation/wormhole/blob/02f468f776f6c8b8d20c1cb5d192818c4fc20c75/node/pkg/processor/processor.go#L323